server/boards/app/files.go

```diff
@@ -175,3 +190,150 @@ func (a *App) MoveFile(channelID, teamID, boardID, filename string) error {
 	}
 	return nil
 }
+
+func (a *App) CopyAndUpdateCardFiles(boardID, userID string, blocks []*model.Block, asTemplate bool) error {
+	if err := a.CopyCardFiles(boardID, blocks, asTemplate); err != nil {
+		a.logger.Error("Could not copy files while duplicating board", mlog.String("BoardID", boardID), mlog.Err(err))
+	}
+
+	// blocks now has updated file ids for any blocks containing files.  We need to update the database for them.
+	blockIDs := make([]string, 0)
+	blockPatches := make([]model.BlockPatch, 0)
+	for _, block := range blocks {
+		if block.Type == model.TypeImage || block.Type == model.TypeAttachment {
+			fieldName := "fileId"
+			if fileID, ok := block.Fields[fieldName]; ok {
```

Comment:
I don't understand what this is accomplishing. It reads like it's just updating the field to it's existing value. Also deleting another field.

Reply:
This is patching the database with the new filenames. Agree, it's confusing, updated the CopyCardFiles to return a map of the filename changes. Should make it easier to understand.